test/suite/surveyPrompt.test.ts

```diff
@@ -0,0 +1,101 @@
+  });
+
+  teardown(() => {
+    sandbox.restore();
+  });
+
+  suite('shouldShowBanner', () => {
+    test('should show banner when all conditions are met', () => {
+      const surveyPrompt = new SurveyPrompt(extensionContext);
+      sandbox.stub(surveyPrompt, 'tookSurveyRecently').returns(false);
+      sandbox.stub(surveyPrompt, 'tookMostRecentVersionOfSurvey').returns(false);
+      sandbox.stub(SurveyPrompt, 'getRandomInt').returns(1);
+
+      assert.strictEqual(surveyPrompt.shouldShowBanner(), true);
+    });
+
+    test('should not show banner if doNoShowAgain is true', () => {
+      globalState.update(StorageKeys.doNotShowAgain, true);
+      const surveyPrompt = new SurveyPrompt(extensionContext);
+      const recentSpy = sandbox.spy(surveyPrompt, 'tookSurveyRecently');
+      const versionSpy = sandbox.spy(surveyPrompt, 'tookMostRecentVersionOfSurvey');
+      const randomSpy = sandbox.spy(SurveyPrompt, 'getRandomInt');
+
+      assert.strictEqual(surveyPrompt.shouldShowBanner(), false);
+      assert.strictEqual(recentSpy.callCount, 0);
+      assert.strictEqual(versionSpy.callCount, 0);
+      assert.strictEqual(randomSpy.callCount, 0);
+    });
+
+    test('should not show banner if tookSurveyRecently is true', () => {
+      const surveyPrompt = new SurveyPrompt(extensionContext);
+      sandbox.stub(surveyPrompt, 'tookSurveyRecently').returns(true);
+      const versionSpy = sandbox.spy(surveyPrompt, 'tookMostRecentVersionOfSurvey');
+      const randomSpy = sandbox.spy(SurveyPrompt, 'getRandomInt');
+
+      assert.strictEqual(surveyPrompt.shouldShowBanner(), false);
+      assert.strictEqual(versionSpy.callCount, 0);
+      assert.strictEqual(randomSpy.callCount, 0);
+    });
+
+    test('should not show banner if tookMostRecentVersionOfSurvey is true', () => {
+      const surveyPrompt = new SurveyPrompt(extensionContext);
+      sandbox.stub(surveyPrompt, 'tookSurveyRecently').returns(true);
+      sandbox.stub(surveyPrompt, 'tookMostRecentVersionOfSurvey').returns(true);
+      const randomSpy = sandbox.spy(SurveyPrompt, 'getRandomInt');
+
+      assert.strictEqual(surveyPrompt.shouldShowBanner(), false);
+      assert.strictEqual(randomSpy.callCount, 0);
+    });
+
+    test('should not show banner if not in 20%', () => {
+      const surveyPrompt = new SurveyPrompt(extensionContext);
+      sandbox.stub(surveyPrompt, 'tookSurveyRecently').returns(true);
+      sandbox.stub(surveyPrompt, 'tookMostRecentVersionOfSurvey').returns(true);
+      sandbox.stub(SurveyPrompt, 'getRandomInt').returns(30);
+
+      assert.strictEqual(surveyPrompt.shouldShowBanner(), false);
+    });
+  });
+
+  suite('tookMostRecentVersionOfSurvey', () => {
+    test('returns false if survey was never taken', () => {
+      const surveyPrompt = new SurveyPrompt(extensionContext);
+      assert.strictEqual(surveyPrompt.tookMostRecentVersionOfSurvey(), false);
+    });
+
+    test('returns true if older version of survey was taken', () => {
```

Comment:
Nit: I think this needs to be
```suggestion
    test('returns false if older version of survey was taken', () => {
```

Reply:
good catch, updated!